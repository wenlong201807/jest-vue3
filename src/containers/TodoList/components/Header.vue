<template>
  <div class="header">
  <div class="header-content">
    TodoList
    <input
      class="header-input"
      placeholder="TodoItem"
      data-test="input"
      v-model="inputValue"
      @keyup.enter="addTodoItem" />
  </div>
  </div>
</template>

<script>
export default {
  name: 'C-Header',
  // props: {},
  data () {
    return {
      inputValue: ''
    };
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue);
        this.inputValue = '';
      }
    }
  }
};
</script>

<style lang="scss">
.header {
  line-height: 60px;
  background: #666;
}
.header-content {
  width: 100%;
  margin: 0 auto;
  color: #fff;
  font-size: 16px;
  text-align: center;
}
.header-input {
  float: right;
  width: 60%;
  margin-top: 16px;
  line-height: 24px;
  outline: none;
  color: #333;
  text-indent: 10px;
}
</style>
