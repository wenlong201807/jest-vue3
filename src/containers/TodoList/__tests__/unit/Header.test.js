import { shallowMount } from "@vue/test-utils";
import Header from "../../components/Header.vue";

it('Header 样式发生改变，做提示', () => {
  const wrapper = shallowMount(Header);
  expect(wrapper).toMatchSnapshot();
})

it('Header 包含 input 框 ', () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  expect(input.exists()).toBe(true);
})

it('Header 中 input 框初始值内容为空', () => {
  const wrapper = shallowMount(Header);
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe('');
})

it('Header 中 input 框值发生变化，数据应该跟着变', () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue('dell lee');
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe('dell lee');
})

it('Header 中 input 框输入回车，无内容时，无反应', () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue('');
  input.trigger('keyup.enter');
  expect(wrapper.emitted().add).toBeFalsy();
})

it('Header 中 input 框输入回车，有内容时，向外触发事件, 同时清空输入框内容', () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue('has content');
  input.trigger('keyup.enter');
  expect(wrapper.emitted().add).toBeTruthy();
  expect(wrapper.vm.$data.inputValue).toBe('');
})





