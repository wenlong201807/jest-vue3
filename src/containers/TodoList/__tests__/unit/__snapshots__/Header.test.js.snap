// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input placeholder="TodoItem" data-test="input" class="header-input">
  </div>
</div>
`;

exports[`Header 组件 样式发生改变，需做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input placeholder="TodoItem" data-test="input" class="header-input">
  </div>
</div>
`;
